src/DataSets/Forms/Save.component.js

```diff
@@ -8,22 +8,22 @@ import CardText from 'material-ui/Card/CardText';
 const Save = React.createClass({
     mixins: [Translate],
 
-    states: {DATAENTRY: "DATAENTRY", SAVED: "SAVED", SAVE_ERROR: "SAVE_ERROR"},
-
     propTypes: {
         config: React.PropTypes.object,
-        data: React.PropTypes.object,
+        store: React.PropTypes.object,
         state: React.PropTypes.string,
         errors: React.PropTypes.arrayOf(React.PropTypes.string)
     },
 
     render() {
-        const {dataset} = this.props.data;
+        const {dataset} = this.props.store;
         const {state, errors} = this.props;
 
-        if (state == this.states.DATAENTRY) {
+        if (state === "DATAENTRY" || state === "SAVING") {
```

Comment:
Good catch, I was a bit uneasy with this, but wasn't sure if it was worth to have a module/mixin. Probably a module with the constants is enough. Willdo.